webapp/src/index.js

```diff
@@ -27,10 +27,13 @@ class PluginClass {
         // This needs to be called before any API calls below
         Client.setServerRoute(getPluginServerRoute(store.getState()));
 
+        const {data: settings} = await getSettings(store.getState);
         await getConnected(true)(store.dispatch, store.getState);
 
-        registry.registerLeftSidebarHeaderComponent(SidebarHeader);
-        registry.registerBottomTeamSidebarComponent(TeamSidebar);
+        if (settings && settings.left_sidebar_enabled) {
+            registry.registerLeftSidebarHeaderComponent(SidebarHeader);
+            registry.registerBottomTeamSidebarComponent(TeamSidebar);
+        }
```

Comment:
We should have a websocket message that tells the frontend when this value has changed, and `register`/`unregister` based on the new value.

Reply:
@mickmister can you give a reference for this.

Reply:
Look for usage of `gitlab_refresh` in the backend, and how it hooks into the frontend:

https://github.com/mattermost/mattermost-plugin-gitlab/blob/aec82089fa47461d48a72bd71a362b6c2c314d1c/webapp/src/index.js#L44-L46

In order to unregister a component, you need to store the id returned from `registry.registerLeftSidebarHeaderComponent`, and pass the stored id to `registry.unregisterComponent` when necessary.